### PR TITLE
Add new config options for creating websocket via template.

### PR DESCRIPTION
**Pull request by jaycee**

### CODE REVIEW

**Comment by jaycee** [comment]
QA:

Setup the charm with a python package created from https://github.com/juju/juju-gui/pull/1140

```
make package
cp /path/to/jujugui-1.9.4.tar.bz2 releases/jujugui-1.9.4
```

Ensure you are using juju 1.25 or higher and deploy the charm.

```
export JUJU_DEV_FEATURE_FLAGS=jes
juju bootstrap
make deploy
```

Create a second env and switch between the two; make sure that works and the rest of the GUI does as well.

File: server/guiserver/apps.py [hunk 1]
```diff
@@ -89,6 +89,10 @@ def server():
         'jujugui.sandbox': options.sandbox,
         'jujugui.raw': options.jujuguidebug,
         'jujugui.combine': (not options.jujuguidebug),
+        'jujugui.apiAddress': options.apiurl,
+        'jujugui.socketTemplate': '/environment/$uuid/api',
+        'jujugui.jujuCoreVersion': options.jujuversion,
+        'jujugui.uuid': options.uuid
```
Comment (frankban) [inline server/guiserver/apps.py]: Why is uuid required still? It shouldn't, right?

File: server/guiserver/manage.py [hunk 2]
```diff
@@ -137,12 +136,10 @@ def setup():
...
         help='Set to True to run the gui without minifiying or combining '
              'source files.')
-    define(
-        'user', type=str,
-        help='The juju environment user.')
-    define(
-        'password', type=str,
-        help='The juju environment user.')
+    define('user', type=str, help='The juju environment user.')
+    define('password', type=str, help='The juju environment user.')
```
Comment (frankban) [inline server/guiserver/manage.py]: s/user/password

File: server/guiserver/manage.py [hunk 3]
```diff
@@ -1,7 +1,6 @@
 # This file is part of the Juju GUI, which lets users view and manage Juju
 # environments within a graphical interface (https://launchpad.net/juju-gui).
-# Copyright (C) 2013 Canonical Ltd.
-#
+# Copyright (C) 2013 Canonical Ltd.  #
```
Comment (frankban) [inline server/guiserver/manage.py]: Leftover?

**Comment by frankban** [comment]
:+1: with a couple of suggestions. QA ok.
Thank you!

File: server/guiserver/apps.py (hunk 1 above)
Comment (jaycee) [inline server/guiserver/apps.py]: It's used as a means to pass the uuid of the initial environment to the gui so we can calculate the initial websocket.

**Comment by makyo** [comment]
:+1:

**Pull request merged**